datafusion/common/src/scalar.rs

```diff
@@ -132,15 +129,19 @@ impl PartialEq for ScalarValue {
             (Boolean(v1), Boolean(v2)) => v1.eq(v2),
             (Boolean(_), _) => false,
             (Float32(v1), Float32(v2)) => {
-                let v1 = v1.map(OrderedFloat);
-                let v2 = v2.map(OrderedFloat);
-                v1.eq(&v2)
+                // Handle NaN == NaN as true manually like in OrderedFloat
```

Comment:
To be consistent with the hash implementation, this should also use total_cmp. Otherwise two equal values, e.g. +0 and -0, will have different hashes

Reply:
What if 
```
                match (v1, v2) {
                    (Some(f1), Some(f2)) => f1.total_cmp(f2).is_eq(),
                    _ => v1.eq(v2),
                }
```